### ISSUE TITLE
Performance Consummer with no message duplication.

### ISSUE THREAD

**Issue by ghetze** [confluentinc/confluent-kafka-dotnet#1195]
Description
===========
Hi,
I am having a .net core app which consumes in real time messages from a topic and processes them.
Currently i am using the autocommit feature of confluent. (EnableAutoCommit = true)
The problem with this approach is that in case of rebalancing the consummers will process duplicate messages.
As a solution to this we've tested a version on which we commit manually the offset (EnableAutoCommit = false).
The drawback of manually commit after each record is the speed of processing decreased with almoust 50%.
We could increase the speed by adding more applications registered to the same GroupId .
Is there another way to have high speed consumers with no message duplication , or is this the only way ?

```
using (var schemaRegistryClient = new CachedSchemaRegistryClient(schemaRegistryConfig))
            {
                using (_consumer = new ConsumerBuilder<Ignore, byte[]>(config)
                    .SetErrorHandler(ConsumerOnError)
                    .Build())
                {
                    _consumer.Subscribe("test_topic");
                    while (true)
                    {
                        try
                        {
                            var consumeResult = _consumer.Consume(_cts.Token);
                            //here we process the message 
                            }
                            await engine.SendMessage(message);
                            try
                            {
                                _consumer.Commit(consumeResult);
                            }
                            catch (KafkaException e)
                            {
                                _logger.LogError(e, $"Commit error: {e.Error.Reason}");
                            }
                        }
                        catch (ConsumeException e)
                        {
                             _logger.LogError(e, "Exception encountered:");
                        }
                        catch (Exception ex)
                        {
                            _logger.LogError(ex, "Exception encountered:");
                        }
                    }
                }
            }
```

How to reproduce
================

Checklist
=========

Please provide the following information:
 - [ ] A complete (i.e. we can run it), minimal program demonstrating the problem. No need to supply a project file.
 - [x ] Confluent.Kafka nuget version. 1.3.0
 - [x] Apache Kafka version.: 2.30.
 - [ ] Client configuration.
 - [x ] Operating system:  Ubuntu 16.04
 - [ ] Provide logs (with "debug" : "..." as necessary in configuration).
 - [ ] Provide broker log excerpts.
 - [x ] Critical issue.

**Comment by mhowlett** [comment]
`EnableAutoCommit = true` actually gives you neither at least once or at most once semantics - it's a bad default setting. Note: it's not event least-once because the commit to kafka happens in a different thread to your processing loop - it's possible for your application to crash in state where the commit has been sent, but your application has not processed the message.

1.4.0-RC2 includes transactions which will allow you do performant exactly once processing, assuming your processing step is idempotent. example EOS processing here: https://github.com/confluentinc/confluent-kafka-dotnet/tree/1.4.x/examples/Transactions

**Comment by ghetze** [comment]
Hi..... I undestand this will be available in 1.4.0 with transactions ... but I need somethig to work with 1.3.0, as a quick fix.
Is the code I've presented in the description enought to achieve this or is there another way.
Thank you.

**Comment by ghetze** [comment]
Hi again...can you please assist..
